### DISCUSSION TITLE
Error when trying to compile basic example outside of original workspace.

### DISCUSSION THREAD

**Discussion by And329** [coderedart/egui_overlay#35] [Q&A] [answered]
My main.rs is same as basic example
My Cargo.toml is: 
`[package]
name = "dev_ov"
version = "0.1.0"
edition = "2021"

[features]
default = ["three_d"]
three_d = ["dep:egui_render_three_d", "egui_overlay/three_d"]
wgpu = ["dep:egui_render_wgpu", "egui_overlay/wgpu"]

[dependencies]
tracing-subscriber = { version = "0.3", features = ["env-filter"] }
egui = { version = "0.27.2"}
egui_overlay = { version = "0.8.1", default-features = false, features = ["egui_default", "glfw_default"]}
egui_window_glfw_passthrough = { version = "0.8.1"}
egui_render_three_d = { version = "0.8.0", optional = true }
egui_render_wgpu = { version = "0.8.0", optional = true}`

Here's compiler error:
`error[E0053]: method `gui_run` has an incompatible type for trait
  --> src/main.rs:30:23
   |
30 |         egui_context: &egui::Context,
   |                       ^^^^^^^^^^^^^^
   |                       |
   |                       expected `egui::context::Context`, found `egui::Context`
   |                       help: change the parameter type to match the trait: `&egui::context::Context`
   |
   = note: expected signature `fn(&mut HelloWorld, &egui::context::Context, &mut ThreeDBackend, &mut GlfwBackend)`
              found signature `fn(&mut HelloWorld, &egui::Context, &mut ThreeDBackend, &mut GlfwBackend)``

If i change `egui_context: &egui::Context,` to `egui_context: &egui::context::Context,`  compiler gives me this:
`error[E0603]: module `context` is private
   --> src/main.rs:30:30
    |
30  |         egui_context: &egui::context::Context,
    |                              ^^^^^^^ private module
    |
note: the module `context` is defined here
   --> C:\Users\zen\.cargo\registry\src\index.crates.io-6f17d22bba15001f\egui-0.27.2\src\lib.rs:379:1
    |
379 | mod context;
    | ^^^^^^^^^^^
help: consider importing this struct instead
    |
30  |         egui_context: &egui::Context,
    |                        ~~~~~~~~~~~~~

error[E0053]: method `gui_run` has an incompatible type for trait
  --> src/main.rs:30:23
   |
30 |         egui_context: &egui::context::Context,
   |                       ^^^^^^^^^^^^^^^^^^^^^^^
   |                       |
   |                       expected `egui::context::Context`, found `egui::Context`
   |                       help: change the parameter type to match the trait: `&egui::context::Context`
   |
   = note: expected signature `fn(&mut HelloWorld, &egui::context::Context, &mut ThreeDBackend, &mut GlfwBackend)`
              found signature `fn(&mut HelloWorld, &egui::Context, &mut ThreeDBackend, &mut GlfwBackend)``

**Comment by coderedart** [comment, marked as answer]
The egui versions are different. egui_overlay is on version [`0.26.*`](https://crates.io/crates/egui_overlay/0.8.1/dependencies)

Just change line ` egui = { version = "0.27.2"}` to use `version = "0.26"`
